boot/zephyr/serial_adapter.c

```diff
@@ -22,10 +22,25 @@
 #include "bootutil/bootutil_log.h"
 #include <zephyr/usb/usb_device.h>
 
-#if defined(CONFIG_BOOT_SERIAL_UART) && defined(CONFIG_UART_CONSOLE)
-#error Zephyr UART console must been disabled if serial_adapter module is used.
+#if defined(CONFIG_BOOT_SERIAL_UART)
+	#if (DT_NODE_HAS_STATUS(DT_CHOSEN(mcuboot_recovery_uart), okay))
+		#define RECOVERY_UART DT_CHOSEN(mcuboot_recovery_uart)
+	#elif (DT_NODE_HAS_STATUS(DT_ALIAS(mcuboot_recovery_uart), okay))
+		#define RECOVERY_UART DT_ALIAS(mcuboot_recovery_uart)
+	#elif (DT_NODE_HAS_STATUS(DT_ALIAS(recovery_uart), okay))
+		#define RECOVERY_UART DT_ALIAS(recovery_uart)
```

Comment:
Why is there another one?

Reply:
This was to be backwards compatible with device trees we already had.  The general solution could just be chosen.